web/src/remark/util/code-blocks.ts

```diff
@@ -1,36 +1,51 @@
 import type * as md from "mdast";
-import type { VFile } from "vfile";
+import { formatCode, FormatCodeOptions } from "./prettier";
+import { wrapInWordBoundaries } from "./regex";
 
 /**
- * Creates a "check fn" for `unist-util-visit` that
- * checks if a node is a code block with a specific meta flag.
+ * Creates a "check fn" for `unist-util-visit` that checks if a node is a code
+ * block with a specific meta flag.
  */
-export function checkNodeIsCodeWithMeta(metaFlag: RegExp) {
+export function makeCheckForCodeWithMeta(metaFlag: string) {
+  // We wrap the string in word boundaries, so we don't match it as a substring
+  // of another word.
+  const metaFlagRegexp = wrapInWordBoundaries(metaFlag);
+
   return (node: md.Nodes): node is md.Code & { meta: string } =>
-    node.type === "code" && node.meta && metaFlag.test(node.meta);
+    node.type === "code" && node.meta && metaFlagRegexp.test(node.meta);
 }
 
 /**
- * Checks that the code block's language is supported.
- * If not, it throws an error with the block's position
- * (through VFile#fail)
+ * Checks that the code block's language is supported. If not, it throws an error.
  */
-export function assertSupportedLanguage<T extends string>(
-  node: md.Code,
-  file: VFile,
-  supportedLanguages: Set<T>,
-): asserts node is md.Code & { lang: T } {
+export function assertSupportedLanguage<
+  Language extends string,
+  CodeNode extends md.Code,
+>(
+  node: CodeNode,
+  supportedLanguages: readonly Language[],
+): asserts node is CodeNode & { lang: Language } {
   if (!node.lang) {
-    file.fail(
-      `No language specified. Please use one of: ${[...supportedLanguages].join(", ")}`,
-      { place: node.position },
-    );
+    throwLangError("No language specified.", supportedLanguages);
   }
 
-  if (!(supportedLanguages as Set<string>).has(node.lang)) {
-    file.fail(
-      `Unsupported language: ${node.lang}. Please use one of: ${[...supportedLanguages].join(", ")}`,
-      { place: node.position },
-    );
+  if (!(supportedLanguages as readonly string[]).includes(node.lang)) {
+    throwLangError(`Unsupported language: ${node.lang}.`, supportedLanguages);
   }
 }
+
+function throwLangError(
```

Comment:
Curious: why not put this function inside the assertSupportedLanguage function? I guess it might have a bit of performance impact, but I imagine that is negligible, while you potentially get nicer scoping, in the sense that it is clear it is helper function inside of this bigger function and is not intended to be used anywhere else.

This one can though probably stand on its own, so all fine with me if you like this better, but I would be tempted to put it inside, so just curious to hear your reasoning for the opposite.

Reply:
Personal preference, vaguely based on perf. I am generally averse to creating utility functions inside other functions (esp. in the hot path). This used to make sense due to the function having to be reconstructed every time, but probably JS engines are intelligent enough now to work around it.

No strong opinion on my side, but if there's also no strong opinion from your side I'll leave it as-is.